examples/types.cpp

```diff
@@ -45,6 +45,11 @@ struct World
   ~World() { std::cout << "Destroying World with message " << msg << std::endl; }
 };
 
+inline bool operator<(const World &lhs, const World &rhs)
+{
+  return lhs.msg > rhs.msg;
```

Comment:
Actually, I realize now that this needs to be added even if a set of World is never used. The set wrappers should be added only if the type supports `operator<`, I think you have to implement a custom type trait for this. Otherwise we will break things for classes needing e.g. support for `std::vector` but not implementing `<`

Reply:
I have implemented a custom type trait to check if a type can be put in a set. If `T` is a container, I'm recursively checking if the type inside has the `<` operator. This is because `std::vector` has lexicographic comparisons but can only be put in a set if the underlying type has the `<` operator. Now, only these types are wrapped.

I tried using this for a simpler implementation: https://en.cppreference.com/w/cpp/concepts/totally_ordered, but this wasn't working on MacOS for some reason: [`71cea02` (#154)](https://github.com/JuliaInterop/libcxxwrap-julia/pull/154/commits/71cea02606fa4217b49bc66de3baca80ff8df446), so I'm sticking with the trait for now.